config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/environment/EnvironmentBundleUtils.java

```diff
@@ -24,8 +30,23 @@
 
     private static final Logger logger = Logger.getLogger(EnvironmentBundleUtils.class.getName());
 
+    private static String templatizedBundlesFolderPath;
+
     private EnvironmentBundleUtils() {}
 
+    static void setTemplatizedBundlesFolderPath(String folderPath) {
+        templatizedBundlesFolderPath = folderPath;
+    }
+
+    public static Bundle getDeploymentBundle() {
+        if (templatizedBundlesFolderPath == null) {
+            throw new BundleLoadException("Invalid deployment bundle path : " + templatizedBundlesFolderPath);
+        }
+        EntityBundleLoader loader = ConfigBuilderModule.getInjector().getInstance(EntityBundleLoader.class);
+        List<File> deploymentBundleFiles = collectFiles(templatizedBundlesFolderPath, BUNDLE_EXTENSION);
+        return loader.load(deploymentBundleFiles);
```

Comment:
You may want to cache this bundle instance so you don't have to read every time someone call this method.
Actually is used only once, but it will be more used.

Reply:
I think this may be out of scope for now. It should be added to rally as an improvement.

Reply:
You are adding an extra step of loading bundles for every listen port, and you are reloading files from disk every time.
I know this does not happen now but lets imagine someone have 10 listen ports specified and and 10 deployment bundles, none of the listen ports pointing to any service, and in this case you will be reading 10 times the same 10 files for nothing, which goes to 100 I/O operations.
I think it won't be that much work to cache the bundle instance and you avoid 99 IO operations in this case.

if the code isn't that easy ok, but it's probably something we would have to fix soon.

Reply:
Ah, I understand. I'll look into it further.